spark-jobs/src/main/scala/za/co/absa/enceladus/standardization/StandardizationConfig.scala

```diff
@@ -0,0 +1,169 @@
+        val newOptions = config.cobolOptions match {
+          case Some(a) => Some(a.copy(isXcom = value))
+          case None => Some(CobolOptions(isXcom = value))
+        }
+        config.withCobolOptions(newOptions)
+      }).text("Does a mainframe file in COBOL format contain XCOM record headers"),
+
+      opt[Boolean]("cobol-is-text").optional().action((value, config) => {
+        val newOptions = config.cobolOptions match {
+          case Some(a) => Some(a.copy(isText = value))
+          case None => Some(CobolOptions(isText = value))
+        }
+        config.withCobolOptions(newOptions)
+      }).text("Specifies if the mainframe file is ASCII text file"),
+
+      opt[String]("cobol-encoding").optional().action((value, config) => {
+        val newOptions = config.cobolOptions match {
+          case Some(a) => Some(a.copy(encoding = Option(value)))
+          case None => Some(CobolOptions(encoding = Option(value)))
+        }
+        config.withCobolOptions(newOptions)
+      }).text("Specify encoding of mainframe files (ascii or ebcdic)"),
+
+      opt[String]("cobol-trimming-policy").optional().action((value, config) => {
+        val newOptions = config.cobolOptions match {
+          case Some(a) => Some(a.copy(trimmingPolicy = Option(value)))
+          case None => Some(CobolOptions(trimmingPolicy = Option(value)))
+        }
+        config.withCobolOptions(newOptions)
+      }).text("Specify string trimming policy for mainframe files (none, left, right, both)"),
+
+      opt[String]("debug-set-raw-path").optional().hidden().action((value, config) =>
+        config.withRawPathOverride(Some(value)))
+        .text("override the path of the raw data (used internally for performance tests)"),
+
+      checkConfig(config => {
+        def foundCsvField(config: R) = config match {
+          case _ if config.csvDelimiter.isDefined => Some("--delimiter")
+          case _ if config.csvEscape.isDefined => Some("--escape")
+          case _ if config.csvHeader.contains(true) => Some("--header")
+          case _ if config.csvQuote.isDefined => Some("--quote")
+          case _ => None
+        }
+
+        def foundCobolField(cobolOptions: CobolOptions) = cobolOptions match {
+          case _ if cobolOptions.copybook != "" => Some("--copybook")
+          case _ if cobolOptions.encoding.isDefined => Some("--cobol-encoding")
+          case _ if cobolOptions.isXcom => Some("--is-xcom")
+          case _ if cobolOptions.isText => Some("--is-text")
+          case _ if cobolOptions.trimmingPolicy.isDefined => Some("--cobol-trimming-policy")
+          case _ => None
+        }
+
+
+        if (!List("xml", "csv", "json", "cobol").contains(config.rawFormat.toLowerCase) && config.charset.isDefined)
+          failure("The --charset option is supported only for CSV, JSON, XML and COBOL")
+        else if (config.rowTag.isDefined && !config.rawFormat.equalsIgnoreCase("xml"))
```

Comment:
`else if` is a bit misleading since these checks are completely independent.
I'd suggest replacing it with a series of ifs or better, extract each check that can call `fail` in a function.